CMakeLists.txt

```diff
@@ -26,6 +26,15 @@ elseif(${ARCH} STREQUAL "arm")
             set(rust_target thumbv7em-zephyr-eabi)
             set(clang_target thumbv7em-unknown-none-eabi)
         endif()
+    elseif(CONFIG_CPU_CORTEX_M33)
```

Comment:
This could probably match on ARMV8_M_MAINLINE to be more generic. (And the above M4/M7 test could have been ARMV7_M_ARMV8_M_MAINLINE)

Reply:
I'll check for more specific config flags

Reply:
@tylerwhall there are also `CONFIG_ARMV7_M_ARMV8_M_FP=y`, `CONFIG_ARMV8_M_DSP=y`, do they also affect flags or target options ?